library/modules/Textures.cpp

```diff
@@ -72,13 +83,148 @@ SDL_Surface * canonicalize_format(SDL_Surface *src) {
   return tgt;
 }
 
+// register surface in texture raws, get a texpos
+static long add_texture(SDL_Surface* surface) {
+    std::lock_guard<std::mutex> lg_add_texture(g_adding_mutex);
+    auto texpos = enabler->textures.raws.size();
+    enabler->textures.raws.push_back(surface);
+    return texpos;
+}
+
+TexposHandle Textures::loadTexture(SDL_Surface* surface) {
+    if (!surface)
+        return 0; // should be some error, i guess
+
+    auto handle = reinterpret_cast<uintptr_t>(surface); // not the best way?
+    g_handle_to_surface.emplace(handle, surface);
+    surface->refcount++; // prevent destruct on next FreeSurface by game
+    auto texpos = add_texture(surface);
+    g_handle_to_texpos.emplace(handle, texpos);
+    return handle;
+}
+
+std::optional<long> Textures::getTexposByHandle(TexposHandle handle) {
+    if (!handle)
+        return std::nullopt;
+
+    if (auto it = g_handle_to_texpos.find(handle); it != g_handle_to_texpos.end())
+        return it->second;
+
+    if (auto it = g_handle_to_surface.find(handle); it != g_handle_to_surface.end()) {
+      it->second->refcount++; // prevent destruct on next FreeSurface by game
+      auto texpos = add_texture(it->second);
+      g_handle_to_texpos.emplace(handle, texpos);
+      return texpos;
+    }
+
+    return std::nullopt;
+}
+
+static void reset_texpos() {
+    g_handle_to_texpos.clear();
+}
+
+static void reset_surface() {
+    for (auto& entry : g_handle_to_surface) {
+        DFSDL_FreeSurface(entry.second);
+    }
+    g_handle_to_surface.clear();
+}
+
+// reset point on New Game
+struct tracking_stage_new_region : df::viewscreen_new_regionst {
+    typedef df::viewscreen_new_regionst interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(void, logic, ()) {
+        if (this->m_raw_load_stage != this->raw_load_stage) {
+            this->m_raw_load_stage = this->raw_load_stage;
+            if (this->m_raw_load_stage == 1)
+                reset_texpos();
+        }
+        INTERPOSE_NEXT(logic)();
```

Comment:
I've tried different combination of `INTERPOSE_NEXT ` and actual trigger stage/step number. If I place it before step check, then it would be +1 step (for the closeset reset point) comapring to texture cleaning in ghidra listing. Should I do that?

Reply:
Сhecked all the options again. If `INTERPOSE_NEXT` placed after step check, then itpossible to use the same step number which game use. And there is no significant interval with invalid texposes during loading screen. There are only single "blink" in couple of loading screen.